src/memory.cpp

```diff
@@ -90,6 +90,24 @@ expr SingleArrayMemory::store(const expr &f32val,
   return z3::ult(idx, block.numelem) && block.writable;
 }
 
+expr SingleArrayMemory::storeArray(
+  const expr &arr, const expr &bid, const expr &offset, const expr &size) {
+  auto low = offset;
+  auto high = offset + size - 1;
+  auto idx = Index("idx");
+  auto arrayVal = z3::select(arr, idx - low);
+
+  auto block = getMemBlock(bid);
+  auto currentVal = z3::select(block.array, idx);
+  auto cond = z3::ule(low, idx) && z3::ule(idx, high);
+  auto stored = z3::lambda(idx, z3::ite(cond, arrayVal, currentVal));
+  arrayMaps = z3::store(arrayMaps, bid, stored);
+
+  return z3::ule(low, high) && // low <= high (to prevent overflow)
+    z3::ule(high, block.numelem) && // high <= block.numelem
```

Comment:
Is it ule or ult?

Reply:
Oh `ult` is right because it is zero based. Thank you for correction!